### ISSUE TITLE
NeoDemo Metal doesn't work on Intel Macbook

### ISSUE THREAD

**Issue by elvodqa** [veldrid/veldrid#518]
OpenGL works fine (excluding the horrendous performance) but when switched to Metal backend, the window is black and occasionally white if you angle the camera to certain positions.

**Comment by tictac1234** [comment]
Totally, I am suffering from the same Issue in Intel iMac as well (2019 model).

But, Does it work on the latest M1 Mac?

**Comment by partyalarm** [comment]
The libsdl2.dylib included with Veldrid.SDL2 is x86_x64 and does not work on arm64 (m1).
I installed sdl2 with homebrew and replaced libsdl2.dylib.
But this time, libveldrid-spirv bundled with Veldrid.SPIRV is x86_x64, so it does not work on arm64 (m1).

**Comment by PaperPrototype** [comment]
Having the same issue
<img width="1297" alt="Screenshot 2024-06-30 at 10 30 31 PM" src="https://github.com/veldrid/veldrid/assets/48071553/6f7283e8-a3e4-4059-bec3-96fbf22940d6">

**Comment by tictac1234** [comment]
Why is this being discussed at all?
Is it just our environment that is not working...?
ummm...

**Comment by elvodqa** [comment]
It's just a demo tho lol. Metal backend of Veldrid is solid enough. Or you can use ppy's fork.

**Comment by tictac1234** [comment]
Thank you for the response.
Is the demo just not running on the latest version ?
Does Metal itself work well ?

**Comment by elvodqa** [comment]
Yes it does. You can use ppy's fork just so there might some changes but even this repo has a great Metal support. Unfortunately there is no Metal 3 but I don't think that's gonna be a big issue .

**Comment by PaperPrototype** [comment]
> Why is this being discussed at all? Is it just our environment that is not working...? ummm...

Metal does not work at all, apparently the dylib isn't a dylib at all and instead it is a DLL

Same issue with spir-v... it doesn't have a mac os compatible dylib and Is using a DLL instead

**Comment by PaperPrototype** [comment]
This issue should be renamed to "Veldrid doesn't work at all on mac os"

**Comment by PaperPrototype** [comment]
`@Driver8` from the veldrid discord has managed to get it working for mac as well as solving the MetalBuffer access 

![Screenshot 2024-07-08 at 1 52 31 AM](https://github.com/veldrid/veldrid/assets/48071553/c0b061aa-2ed9-4e73-be33-8cc1e57042cc)

> Veldrid: https://github.com/seanspicer/veldrid/tree/sas/integrate-ppy-techpizza
> Veldrid-SPIRV: https://github.com/seanspicer/veldrid-spirv/tree/sas/integrate-ppy-techpizza
> 
> Both of these should build on arm64 Macs with no errors and tests should pass.   NeoDemo will fail as illustrated above.

**Comment by PaperPrototype** [comment]
Used Driver8's repo as a drag and drop fix to get [prowl engine](https://github.com/ProwlEngine/Prowl) working on MacOS arm (prowl is using its own fork of Veldrid)

**Comment by partyalarm** [comment]
> Used Driver8's repo as a drag and drop fix to get [prowl engine](https://github.com/ProwlEngine/Prowl) working on MacOS arm (prowl is using its own fork of Veldrid)

Excellent.
Worked on my old MacBook Pro 13 Early 2015.
